test/test_internet.py

```diff
@@ -9,22 +9,61 @@
 import pytest
 import warnings
 from urllib.error import URLError
+from urllib.request import Request
 
 import torchvision.datasets.utils as utils
 from common_utils import get_tmp_dir
 
 
-class TestDatasetUtils:
+@pytest.fixture
+def patch_url_redirection(mocker):
+    class Response:
```

Comment:
could we use the Response class from requests (if there's such a thing?)

Reply:
Apart from the [docs stating](https://docs.python.org/3/library/http.client.html#http.client.HTTPResponse):

> Not instantiated directly by user.

we do not need anything apart from the `url` attribute. Thus, in the spirit of the other suggestions, I would keep it a custom object until we need something more.

Reply:
And if you mean the `requests` package by

> Response class from requests

than no. Since we are using `urllib` here, classes from `requests` will be as foreign as a custom object.

Reply:
I got a bit mixed up, I meant use the class which is returned by `urllib.request.urlopen`. But if that class is any "harder" to instanciate than the custom `Response` you wrote, I agree it's not worth it